File: src/main/java/com/alex323glo/sandbox/rotator/jimpl/Runner.java
package com.alex323glo.sandbox.rotator.jimpl;


public class Runner {
    private static final String TESTED_STRING = "abc(de(fa)ac(kl))gh(123)45";
    private static final String EXPECTED_RESULT = "abcklcafaedgh32145";

    public static void main(String[] args) {
        String rotatedString = Rotator.rotateString(TESTED_STRING);

        System.out.println("In:        \"" + TESTED_STRING + "\"");
        System.out.println("Out:       \"" + rotatedString + "\"");
        System.out.println("EXPECTED:  \"" +EXPECTED_RESULT + "\"");
        System.out.println("Actual result is equal to expected: " + (EXPECTED_RESULT.equals(rotatedString)));
    }
}

File: src/main/kotlin/com/alex323glo/sandbox/rotator/kimpl/procedural/Runner.kt
package com.alex323glo.sandbox.rotator.kimpl.procedural

const val TESTED_STRING = "abc(de(fa)ac(kl))gh(123)45"
const val EXPECTED_RESULT = "abcklcafaedgh32145"

fun main() {

    val rotatedString = rotateString(TESTED_STRING)

    println("In:        \"$TESTED_STRING\"")
    println("Out:       \"$rotatedString\"")
    println("EXPECTED:  \"$EXPECTED_RESULT\"")
    println("Actual result is equal to expected: ${EXPECTED_RESULT == rotatedString}")
}